worker/src/test/java/org/verapdf/webapp/worker/listener/ExecutableTaskWorkerTests.java

```diff
@@ -0,0 +1,555 @@
+package org.verapdf.webapp.worker.listener;
+
+import org.apache.commons.io.FileUtils;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentMatcher;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.core.io.FileSystemResource;
+import org.springframework.core.io.Resource;
+import org.springframework.http.HttpStatus;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.web.client.RestClientResponseException;
+import org.verapdf.webapp.localstorageservice.client.service.LocalStorageServiceClient;
+import org.verapdf.webapp.localstorageservice.model.dto.StoredFileDTO;
+import org.verapdf.webapp.queueclient.listener.QueueListener;
+import org.verapdf.webapp.queueclient.sender.QueueSender;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.*;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.Arrays;
+import java.util.UUID;
+
+@ActiveProfiles("test")
+@SpringBootTest
+public class ExecutableTaskWorkerTests {
+
+	@Autowired
+	private ExecutableTaskWorker executableTaskWorker;
+
+	@Autowired
+	private File workingDir;
+
+	@MockBean
+	private LocalStorageServiceClient localStorageServiceClient;
+
+	@MockBean
+	private QueueSender queueSender;
+
+	@MockBean
+	private QueueListener queueListener;
+
+	@AfterEach
+	public void checkWorkingDirectory() {
+		assertTrue(workingDir.listFiles().length == 0);
+		Mockito.verifyNoMoreInteractions(localStorageServiceClient, queueSender);
+		Mockito.validateMockitoUsage();
+	}
+
+	@Test
+	public void processTest() throws URISyntaxException, IOException {
+		UUID fileId = UUID.fromString("07b17d5d-1010-4980-b280-94129cb13838");
+
+		Mockito.doReturn(createDescriptor(fileId))
+				.when(localStorageServiceClient)
+				.getFileDescriptorById(fileId);
+		Mockito.doReturn(createResource())
+				.when(localStorageServiceClient)
+				.getFileResourceById(fileId);
+		byte[] correctReportBytes = getClass().getResourceAsStream(
```

Comment:
1. naming (better use expected instead of `correct`
2. lets compare by json content (similar to validation tests) to ensure that in future we won't get unstable test depending on fields/array order

Reply:
the same valid for all further similar cases